napari_plugin_manager/base_qt_plugin_dialog.py

```diff
@@ -0,0 +1,1671 @@
+import contextlib
+import os
+import webbrowser
+from functools import partial
+from typing import (
+    Dict,
+    List,
+    Literal,
+    NamedTuple,
+    Optional,
+    Protocol,
+    Sequence,
+    Tuple,
+)
+
+from packaging.version import parse as parse_version
+from qtpy.QtCore import QSize, Qt, QTimer, Signal, Slot
+from qtpy.QtGui import (
+    QAction,
+    QActionGroup,
+    QFont,
+    QKeySequence,
+    QShortcut,
+)
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QComboBox,
+    QDialog,
+    QFrame,
+    QGridLayout,
+    QHBoxLayout,
+    QLabel,
+    QLineEdit,
+    QListWidget,
+    QListWidgetItem,
+    QMenu,
+    QPushButton,
+    QSizePolicy,
+    QSplitter,
+    QTextEdit,
+    QToolButton,
+    QVBoxLayout,
+    QWidget,
+)
+from superqt import QCollapsible, QElidingLabel
+
+from napari_plugin_manager.qt_package_installer import (
+    InstallerActions,
+    InstallerQueue,
+    InstallerTools,
+    ProcessFinishedData,
+)
+from napari_plugin_manager.qt_widgets import ClickableLabel
+from napari_plugin_manager.utils import is_conda_package
+
+CONDA = 'Conda'
+PYPI = 'PyPI'
+
+
+class BasePackageMetadata(Protocol):
+    """Protocol class defining the minimum atributtes/properties needed for package metadata."""
+
+    @property
+    def metadata_version(self) -> str:
+        """Metadata version the package metadata class aims to support."""
+
+    @property
+    def name(self) -> str:
+        """Name of the package being represented."""
+
+    @property
+    def version(self) -> str:
+        """Version of the package being represented."""
+
+    @property
+    def summary(self) -> str:
+        """Summary of the package being represented."""
+
+    @property
+    def home_page(self) -> str:
+        """Home page URL of the package being represented."""
+
+    @property
+    def author(self) -> str:
+        """Author information of the package being represented."""
+
+    @property
+    def license(self) -> str:
+        """License information of the package being represented."""
+
+
+class _BasePackageMetadata(NamedTuple):
```

Comment:
Add docstring here about intent of this NamedTuple, why it is private and how it relates to the protocol above. I wonder: can we use a `dataclass` for both things and have a single object?

Reply:
The main reason why I used the Protocol class + Actual class implementation is that when I tried to subclass the basic class implementation while also subclassing `npe2.PackageMetadata` errors appeared. Will check if using `dataclass` instead of inheriting from `NamedTuple` could help with this.

In the meantime, I think a better selection of names can be done. So something like `PackageMetadataProtocol` and `BasePackageMetadata` as well as some info over the doctring as you mention :+1: